### PR TITLE
wrote some initial code

### PR DESCRIPTION
**Pull request by hklerer**

### CODE REVIEW

File: src/main/java/GameOfLife.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+public class GameOfLife
```
[the remark this answers is not in the document]
Comment (andrewoid) [reply src/main/java/GameOfLife.java]: You must be in a package
`klerer.gameoflife`

File: src/main/java/GameOfLife.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+{
+    static int[][] nextGeneration(int grid[][], int rows, int columns) {
```
Comment (andrewoid) [inline src/main/java/GameOfLife.java]: I fail you if you make static methods.

File: src/main/java/klerer/conways/GameOfLife.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package klerer.conways;
+
+public class GameOfLife
+{
+    static int[][] nextGeneration(int[][] grid, int rows, int columns) {
```
Comment (andrewoid) [inline src/main/java/klerer/conways/GameOfLife.java]: You are not allowed to make static methods.

File: src/main/java/klerer/gameoflife/GameOfLife.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package klerer.gameoflife;
+
+public class GameOfLife {
+    int[][] nextGeneration(int[][] grid, int rows, int columns) {
```
Comment (andrewoid) [inline src/main/java/klerer/gameoflife/GameOfLife.java]: `grid` should be internal to the GameOfLife class.

File: src/main/java/klerer/gameoflife/GameOfLife.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                        if ((l + i >= 0 && l + i < rows) && (m + j >= 0 && m + j < columns)) {
+                            aliveNeighbours += grid[l + i][m + j];
+                        }
+                    }
+                }
+
+                aliveNeighbours -= grid[l][m];
+
+                if ((grid[l][m] == 1) && (aliveNeighbours < 2)) {
+                    future[l][m] = 0;
```
Comment (andrewoid) [inline src/main/java/klerer/gameoflife/GameOfLife.java]: future[l][m] is already 0

File: src/main/java/klerer/gameoflife/GameOfLife.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+                } else if ((grid[l][m] == 1) && (aliveNeighbours > 3)) {
+                    future[l][m] = 0;
```
Comment (andrewoid) [inline src/main/java/klerer/gameoflife/GameOfLife.java]: future[l][m] is already 0

File: src/main/java/klerer/gameoflife/GameOfLife.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
+package klerer.gameoflife;
+
+public class GameOfLife {
+    private int[][] grid;
+
+    public void setGrid(int[][] initialGrid) {
```
Comment (andrewoid) [inline src/main/java/klerer/gameoflife/GameOfLife.java]: If you want to set an initial grid, that's what constructors are for

File: src/main/java/klerer/gameoflife/GameOfLife.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public int[][] getGrid() {
+        return grid;
+    }
+
+    public int[][] nextGeneration() {
+        if (grid == null) {
+            throw new IllegalStateException("Grid has not been initialized");
+        }
```
Comment (andrewoid) [inline src/main/java/klerer/gameoflife/GameOfLife.java]: Grid should never be null

File: src/main/java/klerer/gameoflife/GameOfLife.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                    for (int j = -1; j <= 1; j++) {
+                        if ((l + i >= 0 && l + i < rows) && (m + j >= 0 && m + j < columns) && !(i == 0 && j == 0)) {
+                            aliveNeighbours += grid[l + i][m + j];
+                        }
+                    }
+                }
+
+                if (grid[l][m] == 1) {
+                    if (aliveNeighbours < 2 || aliveNeighbours > 3) {
+                        future[l][m] = 0;
```
Comment (andrewoid) [inline src/main/java/klerer/gameoflife/GameOfLife.java]: It's already 0

**Review by andrewoid — APPROVED** [review]

**Comment by hklerer** [comment]
i did not see that you approved my pr and I made changes. Should I undo those changes and merge?

**Comment by andrewoid** [comment]
> i did not see that you approved my pr and I made changes. Should I undo those changes and merge?

I approved it in anticipation of you fixing that one issue.
You can merge it with the change you made.

**Pull request merged**